extensions/wikia/ArticleVideo/bower_components/html5-skin/js/views/adScreen.js

```diff
@@ -0,0 +1,194 @@
+    Utils = require('../components/utils'),
+    ResizeMixin = require('../mixins/resizeMixin'),
+    Icon = require('../components/icon');
+
+var AdScreen = React.createClass({
+  mixins: [ResizeMixin],
+
+  getInitialState: function() {
+    this.isMobile = this.props.controller.state.isMobile;
+    return {
+      controlBarVisible: true,
+      timer: null
+    };
+  },
+
+  componentDidMount: function () {
+    //for mobile or desktop fullscreen, hide control bar after 3 seconds
+    if (this.isMobile || this.props.fullscreen) {
+      this.props.controller.startHideControlBarTimer();
+    }
+  },
+
+  componentWillUpdate: function(nextProps) {
+    if(nextProps) {
+      if (nextProps.controller.state.controlBarVisible == false && this.state.controlBarVisible == true) {
+        this.hideControlBar();
+      }
+
+      if (nextProps.controller.state.controlBarVisible == true && this.state.controlBarVisible == false) {
+        this.showControlBar();
+      }
+
+      if(!this.props.fullscreen && nextProps.fullscreen && this.state.playerState != CONSTANTS.STATE.PAUSE) {
+        this.props.controller.startHideControlBarTimer();
+      }
+      if(this.props.fullscreen && !nextProps.fullscreen && this.isMobile) {
+        this.showControlBar();
+        this.props.controller.startHideControlBarTimer();
+      }
+    }
+  },
+
+  componentWillUnmount: function () {
+    this.props.controller.cancelTimer();
+  },
+
+  handleResize: function() {
+    if (this.isMounted()) {
+      this.props.controller.startHideControlBarTimer();
+    }
+  },
+
+  handleClick: function(event) {
+    event.stopPropagation(); // W3C
+    event.cancelBubble = true; // IE
+
+    this.props.controller.state.accessibilityControlsEnabled = true;
+    if ((event.type == 'click' || !this.isMobile) && !this.props.skinConfig.adScreen.showAdMarquee) {
+      this.props.controller.onAdsClicked(CONSTANTS.AD_CLICK_SOURCE.VIDEO_WINDOW);
+    }
+  },
+
+  handlePlayerClicked: function(event) {
+    if (event.type == 'touchend' || !this.isMobile) {
+      //since mobile would fire both click and touched events,
+      //we need to make sure only one actually does the work
+
+      //since after exiting the full screen, iPhone pauses the video and places an overlay play button in the middle
+      //of the screen (which we can't remove), clicking the screen would start the video.
+      if (Utils.isIPhone() && this.state.playerState == CONSTANTS.STATE.PAUSE) {
+        this.props.controller.togglePlayPause();
+      }
+      else {
+        event.stopPropagation(); // W3C
+        event.cancelBubble = true; // IE
+        this.props.controller.onAdsClicked(CONSTANTS.AD_CLICK_SOURCE.VIDEO_WINDOW);
+      }
+    }
+  },
+
+  showControlBar: function() {
+    this.setState({controlBarVisible: true});
+    this.props.controller.showControlBar();
+  },
+
+  hideControlBar: function(event) {
+    if (this.props.controlBarAutoHide == true && !(this.isMobile && event)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "true" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1125)